app/services/rating_services.rb

```diff
@@ -0,0 +1,37 @@
+module RatingServices
+  class ArticleRatingCalculator < ApplicationService
+    def initialize(article)
+      @article = article
+    end
+
+    def call
+      article_comments = @article.comments
+      positive = @article.comments.filter { |comment| comment.sentiment == 'Positive' }
```

Comment:
instead of using `filter` several times, use `reduce` or `each_with_object`:
```
filtered_comments = @article.comments.reduce({ positive: [], negative: [], neutral: [] }) do
end
```
and then use these keys where needed, eg `filtered_comments.positive`